src/timeseriesflattenerv2/aggregators.py

```diff
@@ -1,27 +1,93 @@
+import datetime as dt
+
 import polars as pl
+from attr import dataclass
 
 from timeseriesflattenerv2.feature_specs import Aggregator
 
 
 class MinAggregator(Aggregator):
+    name = "min"
+
     def __call__(self, column_name: str) -> pl.Expr:
-        value_col_name = f"{column_name}_min"
-        return pl.col(column_name).min().alias(value_col_name)
+        return pl.col(column_name).min().alias(self.new_col_name(column_name))
 
 
 class MaxAggregator(Aggregator):
+    name = "max"
+
     def __call__(self, column_name: str) -> pl.Expr:
-        value_col_name = f"{column_name}_max"
-        return pl.col(column_name).max().alias(value_col_name)
+        return pl.col(column_name).max().alias(self.new_col_name(column_name))
 
 
 class MeanAggregator(Aggregator):
+    name: str = "mean"
+
     def __call__(self, column_name: str) -> pl.Expr:
-        value_col_name = f"{column_name}_mean"
-        return pl.col(column_name).mean().alias(value_col_name)
+        return pl.col(column_name).mean().alias(self.new_col_name(column_name))
 
 
 class CountAggregator(Aggregator):
+    name: str = "count"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        return pl.col(column_name).count().alias(self.new_col_name(column_name))
+
+
+class EarliestAggregator(Aggregator):
+    name: str = "earliest"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        return pl.col(column_name).first().alias(self.new_col_name(column_name))
+
+
+class LatestAggregator(Aggregator):
+    name: str = "latest"
+
     def __call__(self, column_name: str) -> pl.Expr:
-        value_col_name = f"{column_name}_count"
-        return pl.col(column_name).count().alias(value_col_name)
+        return pl.col(column_name).last().alias(self.new_col_name(column_name))
+
+
+class SumAggregator(Aggregator):
+    name: str = "sum"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        return pl.col(column_name).sum().alias(self.new_col_name(column_name))
+
+
+class VarianceAggregator(Aggregator):
+    name: str = "var"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        return pl.col(column_name).var().alias(self.new_col_name(column_name))
+
+
+class HasValuesAggregator(Aggregator):
+    """Examines whether any values exist in the column. If so, returns 1, else 0."""
+
+    name: str = "bool"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        return (
+            pl.when(pl.col(column_name).is_not_null())
+            .then(1)
+            .otherwise(0)
+            .cast(pl.Boolean)
+            .any()
+            .alias(self.new_col_name(column_name))
+        )
+
+
+@dataclass(frozen=True)
+class SlopeAggregator(Aggregator):
+    timestamp_col_name: str
+    name: str = "slope"
+
+    def __call__(self, column_name: str) -> pl.Expr:
+        # Convert to days for the slope. Arbitrarily chosen to be the number of days since 1970-01-01.
+        x_col = (pl.col(self.timestamp_col_name) - dt.datetime(1970, 1, 1)).dt.days()
```

Comment:
Might want to do `pl.col(self.timestamp_col_name).days` to ensure that the format is days as well. Since we allow an arbitrary timedelta as lookbehind people might go to hour or minute resolution in e.g. an ICU context. But the format might not change anything in this case